collector/container.go

```diff
@@ -0,0 +1,272 @@
+			prometheus.BuildFQName(Namespace, subsystem, "cpu_usage_seconds_usermode"),
+			"Run Time in User mode in Seconds",
+			[]string{"container_id"},
+			nil,
+		),
+		RuntimeKernel: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "cpu_usage_seconds_kernelmode"),
+			"Run time in Kernel mode in Seconds",
+			[]string{"container_id"},
+			nil,
+		),
+		BytesReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_receive_bytes_total"),
+			"Bytes Received on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+		BytesSent: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_transmit_bytes_total"),
+			"Bytes Sent on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+		PacketsReceived: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_receive_packets_total"),
+			"Packets Received on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+		PacketsSent: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_transmit_packets_total"),
+			"Packets Sent on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+		DroppedPacketsIncoming: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_receive_packets_dropped_total"),
+			"Dropped Incoming Packets on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+		DroppedPacketsOutgoing: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "network_transmit_packets_dropped_total"),
+			"Dropped Outgoing Packets on Interface",
+			[]string{"container_id", "interface"},
+			nil,
+		),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *ContainerMetricsCollector) Collect(ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ch); err != nil {
+		log.Error("failed collecting ContainerMetricsCollector metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+func (c *ContainerMetricsCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+
+	// Types Container is passed to get the containers compute systems only
+	containers, err := hcsshim.GetContainers(hcsshim.ComputeSystemQuery{Types: []string{"Container"}})
+	if err != nil {
+		log.Error("Err in Getting containers:", err)
+		return nil, err
+	}
+
+	count := len(containers)
+
+	if count == 0 {
+		return nil, nil
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ContainersCount,
+		prometheus.GaugeValue,
+		float64(count),
+	)
+
+	for _, containerDetails := range containers {
+		containerId := containerDetails.ID
+
+		container, err := hcsshim.OpenContainer(containerId)
+		if err != nil {
+			log.Error("err in opening container: ", containerId)
+			continue
+		}
+
+		cstats, err := container.Statistics()
+		if err != nil {
+			log.Error("err in fetching container Statistics: ", containerId)
```

Comment:
I think we want the error logged as well

Reply:
added err.